packages/snap/src/rpc/vc/createVP.ts

```diff
@@ -1,24 +1,67 @@
 import type { CreateVPRequestParams } from '@blockchain-lab-um/masca-types';
 import { BIP44CoinTypeNode } from '@metamask/key-tree';
-import type { VerifiablePresentation } from '@veramo/core';
+import type {
+  UnsignedPresentation,
+  VerifiablePresentation,
+  W3CVerifiableCredential,
+} from '@veramo/core';
 
 import type { ApiParams } from '../../interfaces';
+import { getCurrentDidIdentifier } from '../../utils/didUtils';
 import { veramoCreateVP } from '../../utils/veramoUtils';
 
+async function createUnsignedVP(params: {
+  vcs: W3CVerifiableCredential[];
+  did: string;
+}): Promise<UnsignedPresentation> {
+  const { vcs, did } = params;
+  const canonicalizedVcs = vcs.map((vc) => {
+    // code from
+    // https://github.com/uport-project/veramo/blob/2ce705680173174e7399c4d0607b67b7303c6c97/packages/credential-eip712/src/agent/CredentialEIP712.ts#L215
+    if (typeof vc === 'string') {
+      return vc;
+    }
+    if (vc.proof.jwt) {
+      return vc.proof.jwt as string;
+    }
+    return JSON.stringify(vc);
+  });
+
+  const unsignedVp: UnsignedPresentation = {
+    holder: did,
+    verifiableCredential: canonicalizedVcs,
+    type: ['VerifiablePresentation', 'Custom'],
+    '@context': ['https://www.w3.org/2018/credentials/v1'],
+    issuanceDate: new Date().toISOString(),
+  };
+  return unsignedVp;
+}
+
 export async function createVP(
   params: ApiParams,
   createVPParams: CreateVPRequestParams
 ): Promise<VerifiablePresentation> {
```

Comment:
I think its better to return `UnsignedPresentation | VerifiablePresentation`

Reply:
Also did the same for `createVC`.